### DISCUSSION TITLE
Email providers compatibility & styling issues

### DISCUSSION THREAD

**Discussion by saharhagekr** [GrapesJS/grapesjs#5769] [Q&A]
Hello,
i have a question
I am using grapesjs  (in my vuejs project), in order to let customers build emails. as i finished building the editor and all the framework, and when we got to testing phase, i noticed some inconsistencies when sending the email. some stuff were because of some email providers incompatibilities or some stuff that they do not support.
for example:
1- flexbox properties. i built a lot of components and blocks based on flex positions, but the properties are not being applied on some email providers like gmail.
2- videos: iFrames are apparently not supported by some providers.
3- media queries: while everything design-related is under a media query, depending on the screen size, any styling under a media query are getting ignored by some email providers.

i know here the problem is coming from support and i am not sure if there is anything i can do about it.

is there any possible solution? does anyone encounter this problem before?

**Comment by mosh-tudor** [comment]
> is there any possible solution? does anyone encounter this problem before?

You should create components and emails using widely supported styles.

- flexbox
  - tables are still recommended for emails :-(
- videos: iFrames
  - use images with a fake play button that links to the video page
- media queries
  - this should work fine if the code is written in the `<<correct>>` way - https://www.caniemail.com/search/?s=media
